src/metabase/models/table.clj

```diff
@@ -296,3 +297,117 @@
+                  ;; why not make this a search term?
+                  :description
+                  ;; bookmark (inserted just-in-time from join for ranking and rendering)
+                  :location]
+
+   :created-at   true
+   :updated-at   false
+
+   :skip         {:namespace [:not nil]}
+
+   ;; depends on the current user, used for rendering and ranking
+   :bookmark     [:model/CollectionBookmark [:and
+                                             [:= :bookmark.collection_id :this.id]
+                                             ;; a magical alias, :current_user (or perhaps this clause can be implicit)
+                                             [:= :bookmark.user_id :current_user/id]]]})
+
+(search/define-spec "card"
+  {:model        :model/Card
+   :context      {:collection_id :collection_id
+                  :database_id   nil
+                  :table_id      nil}
+   :archived     :archived
+   ;; TODO :dataset_query if native
+   :search-terms [:name :description]
+
+   :render-terms [[:collection.name :collection_name]
+                  [:collection.type :collection_type]
+                  [:collection.location :collection_location]
+                  [:collection.authority_level :collection_authority_level]
+                  :archived_directly
+                  :collection_position
+                  :creator_id
+                  ;; dashboard count (TODO oh, best query this when rendering)
+                  [:r.timestamp :last_edited_at]
+                  [:r.user_id :last_editor_id]
+                  [:mr.status :moderated_status]
+                  :display
+                  :dataset_query]
+
+   :bookmark     [:model/CardBookmark [:and
+                                       [:= :bookmark.card_id :this.id]
+                                       [:= :bookmark.user_id :current_user/id]]]
+
+   :skip         {:this.type            [:not "question"]
+                  :collection.namespace [:not nil]}
+
+   :joins        {:collection [:model/Collection [:= :collection.id :this.collection_id]]
+                  :r          [:model/Revision [:and
+                                                [:= :r.model_id :this.id]
+                                                ;; Interesting for inversion - this is a condition on whether to do the update.
+                                                ;; For now let's just swallow up spurious updates (should be 2x amplification)
+                                                [:= :r.most_recent true]
+                                                [:= :r.model "Card"]]]
+                  :mr         [:model/ModerationReview [:and
+                                                        [:= :mr.moderated_item_type "card"]
+                                                        [:= :mr.moderated_item_id :this.id]
+                                                        [:= :mr.most_recent true]]]}
+
+   :created-at   true
+   :updated-at   true})
```

Comment:
do we really need these 2 keys? I only see it used in `select` so maybe it could be defined in `render-terms`?

Reply:
same with archived

Reply:
These will be used for determining whether the search-model is compatible with a time range filter on the results, which it's good to know statically. I don't like putting them in render-terms, since they will have first-class columns as well, but context can work, I'd just have to decide whether to have the "true" magic value there (I already have magic for false / nil so I think yes).

Reply:
Move to :attrs (which is the new name for :context)